src/pprof_export.jl

```diff
@@ -0,0 +1,249 @@
+
+# Keyword Arguments
+- `web::Bool`: Whether to launch the `go tool pprof` interactive webserver for viewing results.
+- `webhost::AbstractString`: If using `web`, which host to launch the webserver on.
+- `webport::Integer`: If using `web`, which port to launch the webserver on.
+- `out::String`: Filename for output.
+- `from_c::Bool`: If `false`, exclude frames that come from from_c. Defaults to `true`.
+- `drop_frames`: frames with function_name fully matching regexp string will be dropped from the samples,
+                 along with their successors.
+- `keep_frames`: frames with function_name fully matching regexp string will be kept, even if it matches drop_functions.
+- `ui_relative_percentages`: Passes `-relative_percentages` to pprof. Causes nodes
+  ignored/hidden through the web UI to be ignored from totals when computing percentages.
+"""
+function _pprof(top_level_path::String, @nospecialize(obj),
+               ;
+               web::Bool = true,
+               webhost::AbstractString = "localhost",
+               webport::Integer = 57599,
+               out::AbstractString = "mem-inspect-profile.pb.gz",
+               from_c::Bool = true,
+               drop_frames::Union{Nothing, AbstractString} = nothing,
+               keep_frames::Union{Nothing, AbstractString} = nothing,
+               ui_relative_percentages::Bool = true,
+            )
+    field_summary = MemorySummarySize.summarysize(obj, parentname=top_level_path)
+
+    period = UInt64(0x1)
+
+    @assert !isempty(basename(out)) "`out=` must specify a file path to write to. Got unexpected: '$out'"
+    if !endswith(out, ".pb.gz")
+        out = "$out.pb.gz"
+        @info "Writing output to $out"
+    end
+
+    string_table = OrderedDict{AbstractString, Int64}()
+    enter!(string) = _enter!(string_table, string)
+    enter!(::Nothing) = _enter!(string_table, "nothing")
+    ValueType!(_type, unit) = ValueType(_type = enter!(_type), unit = enter!(unit))
+
+    # Setup:
+    enter!("")  # NOTE: pprof requires first entry to be ""
+    # Functions need a uid, we'll use the pointer for the method instance
+    seen_funcs = Set{UInt64}()
+    funcs = Dict{UInt64, Function}()
+
+    seen_locs = Set{UInt64}()
+    locs  = Dict{UInt64, Location}()
+
+    sample_type = [
+        ValueType!("instances", "count"), # Mandatory
+        ValueType!("size", "bytes")
+    ]
+
+    prof = PProfile(
+        sample = [], location = [], _function = [],
+        mapping = [], string_table = [],
+        sample_type = sample_type, default_sample_type = 2, # size
+        period = period, period_type = ValueType!("heap", "bytes")
+    )
+
+    if drop_frames !== nothing
+        prof.drop_frames = enter!(drop_frames)
+    end
+    if keep_frames !== nothing
+        prof.keep_frames = enter!(keep_frames)
+    end
+
+    # Mappings to PProf:
+    #  Type of the element => Function
+    #  instance objectid => IP
+    #  path to the element => Stack Trace
+
```

Comment:
I wasn't exactly sure about this mapping. I'm not sure which things should be Functions vs IPs. I was basically experimenting with it, below, and the current configuration _seems_ to work how i'd expect?